src/cache.rs

```diff
@@ -475,6 +432,60 @@ impl Cache {
     }
 }
 
+fn build_proxy() -> Result<Option<Proxy>> {
+    if let Ok(ref host) = env::var("HTTP_PROXY") {
+        return Ok(Some(Proxy::new(host)?));
+    }
+    if let Ok(ref host) = env::var("HTTPS_PROXY") {
+        return Ok(Some(Proxy::new(host)?));
+    }
+
+    Ok(None)
+}
+
+impl Cache {
+    fn build_client(tls_backend: TlsBackend) -> Result<Agent> {
+        let proxy = build_proxy()?;
+        let mut tls_builder = TlsConfig::builder();
+        tls_builder = match tls_backend {
+            #[cfg(feature = "native-tls")]
+            TlsBackend::NativeTls => tls_builder
+                .provider(TlsProvider::NativeTls)
+                .root_certs(RootCerts::PlatformVerifier),
+            #[cfg(feature = "rustls-with-webpki-roots")]
+            TlsBackend::RustlsWithWebpkiRoots => tls_builder
+                .provider(TlsProvider::Rustls)
+                .root_certs(RootCerts::WebPki),
+            #[cfg(feature = "rustls-with-native-roots")]
+            TlsBackend::RustlsWithNativeRoots => tls_builder
+                .provider(TlsProvider::Rustls)
+                .root_certs(RootCerts::PlatformVerifier),
+        };
+        let config = Agent::config_builder()
+            .user_agent(USER_AGENT)
+            .timeout_global(Some(Duration::from_secs(GLOBAL_DOWNLOAD_TIMEOUT_SECONDS)))
```

Comment:
Same thing, why is this better than the default? I would expect that it just waits until the download is complete, which seems preferable for users with a very slow Internet connection

Reply:
Alright, removed to use the default.